streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/TableSourceNode.java

```diff
@@ -27,31 +29,35 @@
 import org.apache.kafka.streams.state.StoreBuilder;
 import org.apache.kafka.streams.state.TimestampedKeyValueStore;
 
+import java.util.Collection;
 import java.util.Collections;
 
 /**
  * Used to represent either a KTable source or a GlobalKTable source. A boolean flag is used to indicate if this represents a GlobalKTable a {@link
  * org.apache.kafka.streams.kstream.GlobalKTable}
  */
-public class TableSourceNode<K, V> extends StreamSourceNode<K, V> {
+public class TableSourceNode<K, V> extends StreamsGraphNode<Void, Void, K, Change<V>> {
```

Comment:
With the extra type information, it's no longer possible to treat a "KTable source" as a kind of "KStream source".